staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -513,8 +510,7 @@ func makeManagedFieldsUpdater(parser *gvkParser, convert convertor, manager stri
 		}
 
 		updater := merge.Updater{Converter: failConverter{}}
-		version := fieldpath.APIVersion(newObjUnstructured.GroupVersionKind().GroupVersion().String())
-		managed, err = updater.Update(oldObjTyped, newObjTyped, version, managed, manager)
+		managed, err = updater.Update(oldObjTyped, newObjTyped, managed, manager)
```

Comment:
A convenient way to convert the `ManagedFields` directly into/from the proper type would be nice also I think